### PR TITLE
feat(cli): backup policy management

### PR DESCRIPTION
**Pull request by lizzy-0323**
## Summary

cli commands for backup policy management:
- create, list ,delete
- pause, update, resume

### CODE REVIEW

File: internal/cli/cmd/backup/delete.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	cmd := &cobra.Command{
+		Use:     "delete <tenant_name>",
+		Short:   "Delete backup policy for the specified tenant",
+		Long:    "Delete backup policy for the specified tenant",
+		Args:    cobra.ExactArgs(1),
+		PreRunE: o.Parse,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := clients.DeleteTenantBackupPolicy(cmd.Context(), types.NamespacedName{
+				Namespace: o.Namespace,
+				Name:      o.Name + "-backup-policy",
```
Comment (powerfooI) [inline internal/cli/cmd/backup/delete.go:38]: If we want the CLI tool to be compatible with other tools, we can not assume that the name of  backup policy for a tenant is `<tenantName>-backup-policy`. For example, if users create backup policy by applying YAML manifests, the name of them may not be like this. Use labels to locate the backup policy.

File: internal/cli/tenant/list.go [hunk 2]
```diff
@@ -29,6 +29,5 @@ func NewListOptions() *ListOptions {
 }
 
 func (o *ListOptions) AddFlags(cmd *cobra.Command) {
-	// These flags are used for further list options, but not used yet.
-	cmd.Flags().StringVar(&o.ClusterName, FLAG_CLUSTER_NAME, "", "The cluster name tenant belonged to in k8s")
+	cmd.Flags().StringVar(&o.Namespace, FLAG_NAMESPACE, DEFAULT_NAMESPACE, "The cluster name tenant belonged to in k8s, if not set, use the default namespace")
```
Comment (powerfooI) [inline internal/cli/tenant/list.go:32]: The help message seems to be wrong.

**Review by powerfooI — COMMENTED** [review]
How about adding a `get/show` sub command to display backup policy for a specific tenant? The command could be like `cli backup show <tenantName>` and it display the information and backup jobs of the backup policy.

**Comment by lizzy-0323** [comment]
> How about adding a `get/show` sub command to display backup policy for a specific tenant? The command could be like `cli backup show <tenantName>` and it display the information and backup jobs of the backup policy.

I think this is a good suggestion and also matches the other commands, and i will complete it when i have time.

File: internal/cli/backup/create.go [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+		_, err := client.GetClient().ClientSet.CoreV1().Secrets(nn.Namespace).Create(ctx, secret, metav1.CreateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	}
+	// set labels for backup policy
+	backupPolicy.Labels = map[string]string{
+		oceanbaseconst.LabelTenantName:      o.Name,
+		oceanbaseconst.LabelRefUID:          string(tenant.GetObjectMeta().GetUID()),
+		oceanbaseconst.LabelRefBackupPolicy: o.Name + "-backup-policy",
```
[the remark this answers is not in the document]
Comment (lizzy-0323) [reply internal/cli/backup/create.go]: Do all these labels need to be deleted?

File: internal/cli/backup/delete.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	generic.ResourceOption
+	force bool
+}
+
+func NewDeleteOptions() *DeleteOptions {
+	return &DeleteOptions{}
+}
+
+func DeleteTenantBackupPolicy(ctx context.Context, o *DeleteOptions) error {
+	nn := types.NamespacedName{Name: o.Name, Namespace: o.Namespace}
```
Comment (powerfooI) [inline internal/cli/backup/delete.go:36]: This is incorrect. This should accept tenant name, and take it as label to filter existing policy and delete it.

File: internal/cli/backup/create.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (powerfooI) [reply internal/cli/backup/create.go]: `LabelRefBackupPolicy` is useless while the two above it make sense.

File: internal/cli/cmd/backup/show.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+			if len(backupJobList.Items) == 0 {
+				logger.Println("no backup jobs found")
+			}
+			if len(backupJobList.Items) > 0 {
+				sort.Slice(backupJobList.Items, func(i, j int) bool {
+					return backupJobList.Items[i].CreationTimestamp.Before(&backupJobList.Items[j].CreationTimestamp)
+				})
+			}
+			for _, job := range backupJobList.Items {
+				tbLog.Println("JOBNAME \t STATUS \t STARTTIME \t ENDTIME \t")
```
Comment (powerfooI) [inline internal/cli/cmd/backup/show.go:65]: Had better add a `type` field by retrieving label `oceanbaseconst.LabelBackupType` of the job.

Comment (lizzy-0323) [reply internal/cli/cmd/backup/show.go:65]: I agree because `jobtype` has been set to as a flag of this command.

**Review by powerfooI — APPROVED** [review]
LGTM

**Pull request merged**